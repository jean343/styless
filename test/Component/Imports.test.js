import React from 'react';
import renderer from 'react-test-renderer';
import styled, {css} from "styled-components";
import 'jest-styled-components';

test('Support import', () => {
    const Div = styled.div`
        @import "import.less";
        color: @color;
        .foo;
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div color="red"/>).toJSON()).toMatchSnapshot();
});

test('Support import reference', () => {
    const Div = styled.div`
        @import (reference) "import.less";
        color: @color;
        .foo;
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
    expect(renderer.create(<Div color="red"/>).toJSON()).toMatchSnapshot();
});