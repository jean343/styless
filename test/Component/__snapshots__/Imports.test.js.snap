// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support import 1`] = `
.c0 {
  color: green;
  background: #900;
  border: 1px solid green;
}

.c0 .foo {
  background: #900;
  border: 1px solid green;
}

<div
  className="c0"
/>
`;

exports[`Support import 2`] = `
.c0 {
  color: red;
  background: #900;
  border: 1px solid red;
}

.c0 .foo {
  background: #900;
  border: 1px solid red;
}

<div
  className="c0"
  color="red"
/>
`;

exports[`Support import reference 1`] = `
.c0 {
  color: green;
  background: #900;
  border: 1px solid green;
}

<div
  className="c0"
/>
`;

exports[`Support import reference 2`] = `
.c0 {
  color: red;
  background: #900;
  border: 1px solid red;
}

<div
  className="c0"
  color="red"
/>
`;
